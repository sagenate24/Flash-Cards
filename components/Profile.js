import React, { Component } from 'react'
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
  ScrollView
} from 'react-native';
import { Foundation } from '@expo/vector-icons';
import { getProfile } from '../utils/api';
import { connect } from 'react-redux';
import { gray, black } from '../utils/colors';
import { timeToString } from '../utils/helpers';
import Score from './Score';

class Profile extends Component {
  state = {
    image: null,
    profileCover: null,
    userName: null,
    status: null,
  }

  componentDidMount() {
    getProfile()
      .then((results) => {
        console.log(results)
        if (results && results.avatar && results.name && results.profileCover ) {
          this.setState(() => ({
            image: results.avatar,
            userName: results.name,
            profileCover: results.profileCover
          }));
        } else {
          this.setState(() => ({
            image: null,
            userName: null,
            profileCover: null
          }));
        }
      })
  }

  render() {
    const { image, profileCover, status, userName } = this.state;
    const { decks } = this.props;

    if (status === 'denied') {
      return (
        <View style={styles.center}>
          <Foundation name='alert' size={50} />
          <Text>
            You denied your location. You can fix this by visiting your settings and enableing location services for this app.
          </Text>
        </View>
      );
    }

    if (status === 'undetermined') {
      return (
        <View style={styles.center}>
          <Foundation name='alert' size={50} />
          <Text>
            You need to enable location services for this app.
          </Text>
          <TouchableOpacity onPress={this.askPermission} style={styles.button}>
            <Text style={styles.buttonText}>
              Enable
            </Text>
          </TouchableOpacity>
        </View>
      );
    }

    return (
      <ScrollView style={styles.container}>
        <View>
          {profileCover && (
            <Image style={styles.topContent} source={{ uri: profileCover }} />
          )}
        </View>
        <View style={styles.profileHeader}>
          {image && (
            <View>
              <Image style={styles.img} source={{ uri: image }} />
            </View>
          )}
          {userName && (
            <Text style={{ fontSize: 30, color: '#000', padding: 20 }}>{userName}</Text>
          )}
        </View>
        <View style={styles.itemContainer}>
          <Text style={styles.quizScoreTitle}>Lastest Quiz Scores</Text>
          {decks.map((deck) => {

            return (
              <View style={styles.item} key={deck.timeStamp}>
                <View style={[styles.itemBox, {alignItems: 'flex-start', flex: 3}]}>
                  <Text style={{ fontSize: 20 }}>
                    {deck.title}
                  </Text>
                  {deck.questions && deck.questions.length
                    ?
                    <Text style={{ fontSize: 16, color: '#6ed3cf' }}>
                      {deck.questions.length > 1
                        ? deck.questions.length + ` Cards`
                        : deck.questions.length + ` Card`}
                    </Text>
                    :
                    <Text style={{ color: gray, fontSize: 16 }}>EMPTY</Text>
                  }
                </View>
                <View style={[styles.itemBox, { alignItems: 'center', flex: 4}]}>
                  {deck && deck.recentScore
                    ? <Score style={{ margin: 10 }} percent={deck.recentScore} width={3} size={45} />
                    : null
                    }
                </View>
                <View style={[styles.itemBox, {alignItems: 'flex-end', flex: 3}]}>
                {deck && deck.recentScore
                    ? <Text style={{ color: '#1b1b7e', fontSize: 16 }}>{timeToString(deck.timeStamp)}</Text>
                    : <Text style={{ fontSize: 16 }}>Take quiz</Text>
                    }
                </View>
              </View>
            )
          })}
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  topContent: {
    width: '100%',
    height: 150,
    backgroundColor: 'white',
    shadowOpacity: 0.8,
    shadowColor: 'rgba(0,0,0,0.24)',
    shadowOffset: {
      width: 0,
      height: 3,
    },
  },
  profileHeader: {
    alignItems: 'center'
  },
  img: {
    width: 140,
    height: 140,
    resizeMode: 'contain',
    borderRadius: 2,
    borderWidth: 1,
    borderColor: 'gray',
    marginTop: -70,
  },
  quizScoreTitle: {
    fontSize: 20,
    color: black,
    fontWeight: 'bold',
  },
  button: {
    padding: 10,
    backgroundColor: '#1b1b7e',
    alignSelf: 'center',
    borderRadius: 5,
    margin: 20,
  },
  buttonText: {
    color: '#fff',
    fontSize: 20,
  },
  itemContainer: {
    padding: 20,
  },
  item: {
    backgroundColor: '#fff',
    borderRadius: 2,
    padding: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 17,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    shadowOpacity: 0.8,
    shadowColor: 'rgba(0,0,0,0.24)',
    shadowOffset: {
      width: 0,
      height: 3,
    },
  },
  itemBox: {
    width: 50,
  }
});

function mapStateToProps(decks) {
  return {
    decks: Object.values(decks).map((deck) => {
      const { timeStamp } = deck;

      return {
        ...deck,
        timeStamp,
      }
    }).sort((a, b) => b.timeStamp - a.timeStamp)
  }
}

export default connect(mapStateToProps)(Profile);