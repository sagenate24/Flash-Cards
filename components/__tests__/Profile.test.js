import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../Profile';

const mockDecks = [
  {
    title: 'React',
    timeStamp: 1534284894237,
    recentScore: 35,
    questions: [
      {
        question: 'What is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is React?',
        answer: 'A library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfaces',
      },
      {
        question: 'Where do you make Ajax requests in React?',
        answer: 'The componentDidMount lifecycle event',
      },
    ],
  },
  {
    title: 'JavaScript',
    timeStamp: 1534284869329,
    recentScore: 100,
    questions: [
      {
        question: 'What is a closure?The combination of a function and the lexical enviornment with in which that function was declaredThe combination of a functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona functiona function',
        answer: 'The combination of a function and the lexical enviornment with in which that function was declared.The combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declaredThe combination of a function and the lexical enviornment with in which that function was declared',
      },
    ],
  },
];

const mockProfile = {
  username: '',
  avatar: '',
  cover: '',
};
// const { decks, profile, navigation } = this.props;

test('Profile snapShot', () => {
  const snap = renderer.create(
    <Profile.WrappedComponent decks={mockDecks} profile={mockProfile} />,
  ).toJSON();

  expect(snap).toMatchSnapshot();
});
