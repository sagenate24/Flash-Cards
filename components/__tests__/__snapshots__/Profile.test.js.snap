// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapShot 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "height": 150,
            "opacity": 1,
            "shadowColor": "rgba(0,0,0,0.24)",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "gray",
              "fontSize": 16,
              "marginTop": 10,
            }
          }
        >
          Choose cover photo
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#fff",
            "borderRadius": 70,
            "borderWidth": 5,
            "height": 140,
            "marginTop": -70,
            "opacity": 1,
            "width": 140,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "marginTop": 45,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#404040",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
                "fontSize": 18,
                "padding": 20,
              },
            ]
          }
        >
          Create Username
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#404040",
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 20,
              "marginBottom": 10,
            },
          ]
        }
      >
        Lastest Quiz Scores
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 17,
              "padding": 20,
              "shadowColor": "rgba(0,0,0,0.24)",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.8,
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              React
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "flex-start",
                    "flex": 3,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#404040",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginTop": 10,
                  }
                }
              >
                2 Cards
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "center",
                    "flex": 3,
                  },
                ]
              }
            >
              <View
                style={undefined}
              >
                <RNSVGSvgView
                  bbHeight="45"
                  bbWidth="45"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      false,
                      Object {
                        "flex": 0,
                        "height": 45,
                        "width": 45,
                      },
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        0,
                        0,
                        0,
                        1,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        -0.3826834323650895,
                        -0.9238795325112868,
                        0.9238795325112868,
                        -0.3826834323650895,
                        10.32308774671056,
                        51.897666709718465,
                      ]
                    }
                    onLayout={undefined}
                    opacity={1}
                    originX={22.5}
                    originY={22.5}
                    propList={Array []}
                    rotation={247.5}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth="1"
                    x={0}
                    y={0}
                  >
                    <RNSVGPath
                      d="M 7.303175500515019 37.7087647470744 A 21.5 21.5 0 1 0 22.5 1"
                      fill={
                        Array [
                          0,
                          0,
                          0,
                          0,
                          0,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      onLayout={undefined}
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={
                        Array [
                          0,
                          0.9490196078431372,
                          0.9490196078431372,
                          0.9490196078431372,
                          1,
                        ]
                      }
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={1}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth="2"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M 22.5 1 A 21.5 21.5 0 0 0 22.5 1"
                      fill={
                        Array [
                          0,
                          0,
                          0,
                          0,
                          0,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      onLayout={undefined}
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={
                        Array [
                          0,
                          0.9019607843137255,
                          0.2235294117647059,
                          0.27450980392156865,
                          1,
                        ]
                      }
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={1}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth="2"
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20.5,
                      "height": 41,
                      "justifyContent": "center",
                      "left": 2,
                      "position": "absolute",
                      "top": 2,
                      "width": 41,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 10,
                        },
                        Object {
                          "color": "#404040",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    35
                    %
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "flex-end",
                    "flex": 4,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#404040",
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 15,
                      "marginTop": 10,
                    },
                  ]
                }
              >
                Aug 14 2018
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 17,
              "padding": 20,
              "shadowColor": "rgba(0,0,0,0.24)",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.8,
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              JavaScript
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "flex-start",
                    "flex": 3,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#404040",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginTop": 10,
                  }
                }
              >
                1 Card
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "center",
                    "flex": 3,
                  },
                ]
              }
            >
              <View
                style={undefined}
              >
                <RNSVGSvgView
                  bbHeight="45"
                  bbWidth="45"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      false,
                      Object {
                        "flex": 0,
                        "height": 45,
                        "width": 45,
                      },
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        0,
                        0,
                        0,
                        1,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        -0.3826834323650895,
                        -0.9238795325112868,
                        0.9238795325112868,
                        -0.3826834323650895,
                        10.32308774671056,
                        51.897666709718465,
                      ]
                    }
                    onLayout={undefined}
                    opacity={1}
                    originX={22.5}
                    originY={22.5}
                    propList={Array []}
                    rotation={247.5}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth="1"
                    x={0}
                    y={0}
                  >
                    <RNSVGPath
                      d="M 7.303175500515019 37.7087647470744 A 21.5 21.5 0 1 0 22.5 1"
                      fill={
                        Array [
                          0,
                          0,
                          0,
                          0,
                          0,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      onLayout={undefined}
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={
                        Array [
                          0,
                          0.9490196078431372,
                          0.9490196078431372,
                          0.9490196078431372,
                          1,
                        ]
                      }
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={1}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth="2"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M 22.5 1 A 21.5 21.5 0 0 0 22.5 1"
                      fill={
                        Array [
                          0,
                          0,
                          0,
                          0,
                          0,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      onLayout={undefined}
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={
                        Array [
                          0,
                          0.6588235294117647,
                          0.8549019607843137,
                          0.8627450980392157,
                          1,
                        ]
                      }
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={1}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth="2"
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20.5,
                      "height": 41,
                      "justifyContent": "center",
                      "left": 2,
                      "position": "absolute",
                      "top": 2,
                      "width": 41,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 10,
                        },
                        Object {
                          "color": "#404040",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    100
                    %
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 50,
                  },
                  Object {
                    "alignItems": "flex-end",
                    "flex": 4,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#404040",
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 15,
                      "marginTop": 10,
                    },
                  ]
                }
              >
                Aug 14 2018
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
